File: web/src/lib/endpoints.ts
/**
 * 前端 API 模块：与后端路由一一对应的类型化封装。
 */
import { api, withQuery } from "./api";
import type {
  AgentOwnerView,
  AgentView,
  AuditLogView,
  AuthResult,
  AutopilotLogView,
  AutopilotRunResult,
  AutopilotSettingView,
  ChatMessageView,
  ChatSessionView,
  ConnectionView,
  GroupBriefView,
  GroupDetailView,
  GroupMessageView,
  LlmConfigView,
  LlmTestResult,
  McpProbeResult,
  MessageView,
  OwnerProfile,
  Paged,
  PlatformStats,
  SystemInfo,
} from "@/types/api";

/* ---------------- Agent ---------------- */

export interface RegisterAgentPayload {
  name: string;
  slug: string;
  emoji?: string;
  color?: string;
  role: string;
  description?: string;
  industry?: string;
  tags?: string[];
  autoAccept?: boolean;
  online?: boolean;
  owner: { name: string; org: string; title?: string; email?: string };
}

export const agentsApi = {
  register: (payload: RegisterAgentPayload) =>
    api.post<AgentView & { secret: string }>("/api/agents", payload),

  list: (params: {
    q?: string;
    tag?: string;
    industry?: string;
    online?: boolean;
    verified?: boolean;
    sort?: string;
    page?: number;
    pageSize?: number;
  }) => api.get<Paged<AgentView>>(withQuery("/api/agents", params)),

  detail: (slug: string) => api.get<AgentView>(`/api/agents/${slug}`),

  topTags: (limit?: number) => api.get<{ name: string; count: number }[]>(withQuery("/api/agents/tags", { limit })),

  industries: () => api.get<{ name: string; count: number }[]>("/api/agents/industries"),
};

/* ---------------- 对接与消息 ---------------- */

export const connectionsApi = {
  create: (fromAgent: string, toAgent: string) =>
    api.post<{ created: boolean; message: string }>("/api/connections", { fromAgent, toAgent }, { sign: true }),

  list: () => api.get<{ items: ConnectionView[] }>("/api/connections", { sign: true }),

  pending: () => api.get<{ items: { from: { slug: string; name: string }; since: string }[] }>("/api/connections/pending", { sign: true }),

  respond: (peer: string, accept: boolean) =>
    api.post<{ status: string }>("/api/connections/respond", { peer, accept }, { sign: true }),
};

export const messagesApi = {
  send: (fromAgent: string, toAgent: string, text: string, type: "chat" | "task" | "event" = "chat") =>
    api.post<{ message: MessageView; bot: MessageView | null }>(
      "/api/messages",
      { fromAgent, toAgent, text, type },
      { sign: true },
    ),

  history: (peer: string, limit = 50) =>
    api.get<{ messages: MessageView[]; unreadCleared: number }>(withQuery("/api/messages", { peer, limit }), { sign: true }),

  markRead: (peer: string) =>
    api.post<{ cleared: number }>("/api/messages/read", { peer }, { sign: true }),

  unread: () => api.get<{ unread: number }>("/api/messages/unread", { sign: true }),

  a2aSend: (slug: string, text: string, type: "chat" | "task" | "event" = "chat") =>
    api.post<{ message: MessageView; bot: MessageView | null }>(`/api/a2a/${slug}/message`, { text, type }, { sign: true }),
};

/* ---------------- 实时通道 ---------------- */

export const realtimeApi = {
  ticket: (agentSlug?: string) =>
    api.post<{ ticket: string; agentSlug: string; expiresInMs: number }>("/api/realtime/ticket", agentSlug ? { agentSlug } : {}, { sign: true }),
};

/* ---------------- 虚拟伙伴 ---------------- */

export const chatApi = {
  createSession: (title?: string) =>
    api.post<ChatSessionView>("/api/chat/sessions", title ? { title } : {}),

  listSessions: () => api.get<{ items: ChatSessionView[] }>("/api/chat/sessions"),

  getSession: (id: string) =>
    api.get<{ session: ChatSessionView; messages: ChatMessageView[] }>(`/api/chat/sessions/${id}`),

  send: (id: string, text: string) =>
    api.post<{ reply: ChatMessageView; engine: "llm" | "local" }>(`/api/chat/sessions/${id}/messages`, { text }),

  remove: (id: string) => api.del<{ deleted: boolean }>(`/api/chat/sessions/${id}`),
};

/* ---------------- 自主巡航 ---------------- */

export const autopilotApi = {
  settings: () => api.get<AutopilotSettingView>("/api/autopilot/settings", { sign: true }),

  update: (input: Partial<Pick<AutopilotSettingView, "enabled" | "minScore" | "autoGreet" | "industries" | "tags">>) =>
    api.patch<AutopilotSettingView>("/api/autopilot/settings", input, { sign: true }),

  run: () => api.post<AutopilotRunResult>("/api/autopilot/run", {}, { sign: true }),

  logs: (limit = 30) => api.get<{ items: AutopilotLogView[] }>(withQuery("/api/autopilot/logs", { limit }), { sign: true }),
};

/* ---------------- 认证 ---------------- */

export const authApi = {
  register: (payload: { name: string; org: string; title?: string; email: string; password: string }) =>
    api.post<AuthResult>("/api/auth/register", payload),

  login: (email: string, password: string) =>
    api.post<AuthResult>("/api/auth/login", { email, password }),

  logout: () => api.post<{ ok: boolean }>("/api/auth/logout"),

  me: () => api.get<OwnerProfile>("/api/auth/me"),

  myAgents: () => api.get<{ items: AgentOwnerView[] }>("/api/me/agents"),
};

/* ---------------- 系统 ---------------- */

export const systemApi = {
  info: () => api.get<SystemInfo>("/api/system/info"),
  health: () => api.get<{ ok: boolean; service: string }>("/health", { timeoutMs: 4000 }),
};

/* ---------------- MCP 探测 ---------------- */

export const mcpApi = {
  probe: (url: string, headers?: Record<string, string>) =>
    api.post<McpProbeResult>("/api/mcp/probe", { url, headers }, { sign: true }),
};

/* ---------------- 管理端（X-Admin-Token 自动注入） ---------------- */

export const adminApi = {
  llmConfig: () => api.get<LlmConfigView>("/api/admin/llm/config"),

  updateLlmConfig: (input: { baseUrl?: string; model?: string; apiKey?: string }) =>
    api.post<LlmConfigView>("/api/admin/llm/config", input),

  testLlm: () => api.post<LlmTestResult>("/api/admin/llm/test", {}),

  stats: () => api.get<PlatformStats>("/api/admin/stats"),

  audit: (params: { action?: string; page?: number; limit?: number } = {}) =>
    api.get<Paged<AuditLogView>>(withQuery("/api/admin/audit", params)),
};

/* ---------------- 企业助理（「创建企业助理」界面后端） ---------------- */

export interface FaqEntryPayload {
  keywords: string[];
  answer: string;
}

export interface ProductEntryPayload {
  name: string;
  keywords: string[];
  pitch: string;
  priceRange: string;
  followup: string;
}

export interface AssistantProfilePayload {
  capabilities: { secretary: boolean; sales: boolean; ticket: boolean };
  faq: FaqEntryPayload[];
  products: ProductEntryPayload[];
  secretary?: Record<string, unknown>;
  ticket?: Record<string, unknown>;
  escalate?: Record<string, unknown>;
  fallback?: string;
  integrations?: Record<string, string>;
}

export interface AssistantView {
  agent: {
    slug: string;
    name: string;
    emoji: string;
    role: string;
    description: string;
    industry: string;
    tags: string[];
    online: boolean;
    verified: boolean;
  };
  profile: AssistantProfilePayload;
}

export interface CreateAssistantPayload {
  slug?: string;
  name: string;
  emoji?: string;
  color?: string;
  role: string;
  description?: string;
  industry?: string;
  tags?: string[];
  autoAccept?: boolean;
  owner: { name: string; org: string; title?: string; email?: string };
  profile: AssistantProfilePayload;
}

export const assistantsApi = {
  /** 创建企业助理（公开；响应含一次性 secret） */
  create: (payload: CreateAssistantPayload) =>
    api.post<AssistantView & { secret: string }>("/api/assistants", payload),

  /** 查询助理配置（公开） */
  detail: (slug: string) => api.get<AssistantView>(`/api/assistants/${encodeURIComponent(slug)}`),

  /** 更新助理配置（需 Agent 签名 / 主人令牌 / 管理员令牌） */
  update: (slug: string, profile: AssistantProfilePayload) =>
    api.patch<AssistantView>(`/api/assistants/${encodeURIComponent(slug)}`, { profile }, { sign: true }),
};

/* ---------------- 群聊（企业 Agent 微信 · 多 Agent 协作） ---------------- */

export const groupsApi = {
  /** 建群（创建者=当前签名 Agent；memberSlugs 必须与创建者已对接） */
  create: (name: string, memberSlugs: string[]) =>
    api.post<{ id: string; name: string; memberSlugs: string[] }>("/api/groups", { name, memberSlugs }, { sign: true }),

  /** 我的群列表 */
  list: () => api.get<{ items: GroupBriefView[] }>("/api/groups", { sign: true }),

  /** 群详情+成员+最近 50 条消息 */
  detail: (id: string) => api.get<GroupDetailView>(`/api/groups/${encodeURIComponent(id)}`, { sign: true }),

  /** 群主添加成员（须与群主已对接） */
  addMembers: (id: string, agents: string[]) =>
    api.post<{ added: string[] }>(`/api/groups/${encodeURIComponent(id)}/members`, { agents }, { sign: true }),

  /** 发群消息 */
  send: (id: string, text: string) =>
    api.post<GroupMessageView>(`/api/groups/${encodeURIComponent(id)}/messages`, { text }, { sign: true }),

  /** 群消息历史 */
  history: (id: string, limit = 100) =>
    api.get<{ messages: GroupMessageView[] }>(`/api/groups/${encodeURIComponent(id)}/messages?limit=${limit}`, { sign: true }),
};

File: mobile/lib/api/endpoints.dart
import 'client.dart';
import '../models/models.dart';
import '../core/credentials.dart';

class AuthApi {
  AuthApi._();

  /// 一体化注册：账号 + 首个 Agent（一次返回 token + secret）
  static Future<({String token, OwnerProfile owner, String agentSlug, String secret})>
      registerWithAgent({
    required String name,
    required String org,
    required String email,
    required String password,
    String? title,
    required String agentName,
    required String agentRole,
    String? agentIndustry,
  }) async {
    final data = await ApiClient.instance.post('/api/auth/register-with-agent', {
      'name': name,
      'org': org,
      'email': email,
      'password': password,
      if (title != null && title.isNotEmpty) 'title': title,
      'agent': {
        'name': agentName,
        'role': agentRole,
        if (agentIndustry != null && agentIndustry.isNotEmpty) 'industry': agentIndustry,
      },
    });
    final agent = (data['agent'] ?? {}) as Map;
    final owner = OwnerProfile.fromJson((data['owner'] ?? {}) as Map<String, dynamic>);
    await CredentialStore.instance.setOwnerToken(data['token'] as String);
    return (
      token: data['token'] as String,
      owner: owner,
      agentSlug: (agent['slug'] ?? '') as String,
      secret: (data['secret'] ?? '') as String,
    );
  }

  static Future<OwnerProfile> login(String email, String password) async {
    final data = await ApiClient.instance
        .post('/api/auth/login', {'email': email, 'password': password});
    final owner = OwnerProfile.fromJson((data['owner'] ?? {}) as Map<String, dynamic>);
    await CredentialStore.instance.setOwnerToken(data['token'] as String);
    return owner;
  }

  /// 当前登录主人（无状态回显）
  static Future<OwnerProfile> me() async {
    final data = await ApiClient.instance.ownerRequest('GET', '/api/auth/me');
    return OwnerProfile.fromJson(data);
  }

  /// 名下 Agent 列表（owner）
  static Future<List<Map<String, dynamic>>> myAgents() async {
    final data = await ApiClient.instance.ownerRequest('GET', '/api/me/agents');
    return ((data['items'] ?? []) as List).cast<Map<String, dynamic>>();
  }
}

class AgentsApi {
  AgentsApi._();

  /// 广场：多维检索
  static Future<List<AgentView>> list({
    String? q,
    String? tag,
    String? industry,
    bool? online,
    int page = 1,
  }) async {
    final data = await ApiClient.instance.get('/api/agents', query: {
      if (q != null && q.isNotEmpty) 'q': q,
      if (tag != null && tag.isNotEmpty) 'tag': tag,
      if (industry != null && industry.isNotEmpty) 'industry': industry,
      if (online != null) 'online': online ? 'true' : 'false',
      'page': '$page',
    });
    return ((data['items'] ?? []) as List)
        .map((e) => AgentView.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  /// 公开详情
  static Future<AgentView> detail(String slug) async {
    final data = await ApiClient.instance.get('/api/agents/$slug');
    return AgentView.fromJson(data);
  }

  /// 注册 Agent（访客快速创建）
  static Future<({String slug, String secret})> register({
    required String name,
    required String slug,
    required String role,
    required String ownerName,
    required String ownerOrg,
  }) async {
    final data = await ApiClient.instance.post('/api/agents', {
      'name': name,
      'slug': slug,
      'role': role,
      'autoAccept': true,
      'owner': {'name': ownerName, 'org': ownerOrg},
    });
    return (slug: data['slug'] as String, secret: data['secret'] as String);
  }

  /// 某 Agent 的动态（公开）
  static Future<List<MomentView>> moments(String slug, {int limit = 10}) async {
    final data = await ApiClient.instance.get('/api/agents/$slug/moments', query: {'limit': '$limit'});
    return ((data['items'] ?? []) as List)
        .map((e) => MomentView.fromJson(e as Map<String, dynamic>))
        .toList();
  }
}

class ConnectionsApi {
  ConnectionsApi._();

  /// 对接（autoAccept 即时成功，否则 pending）
  static Future<({bool created, String message})> create(String fromAgent, String toAgent) async {
    final data = await ApiClient.instance
        .agentRequest('POST', '/api/connections', body: {'fromAgent': fromAgent, 'toAgent': toAgent});
    return (created: (data['created'] ?? false) as bool, message: (data['message'] ?? '') as String);
  }

  /// 我的伙伴（通讯录）
  static Future<List<ConnectionView>> list() async {
    final data = await ApiClient.instance.agentRequest('GET', '/api/connections');
    return ((data['items'] ?? []) as List)
        .map((e) => ConnectionView.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<List<Map<String, dynamic>>> pending() async {
    final data = await ApiClient.instance.agentRequest('GET', '/api/connections/pending');
    return ((data['items'] ?? []) as List).cast<Map<String, dynamic>>();
  }

  static Future<String> respond(String peer, bool accept) async {
    final data = await ApiClient.instance
        .agentRequest('POST', '/api/connections/respond', body: {'peer': peer, 'accept': accept});
    return (data['status'] ?? '') as String;
  }
}

class MessagesApi {
  MessagesApi._();

  /// 发消息（bot 回执由后端生成）
  static Future<({MessageView message, MessageView? bot})> send(
      String fromAgent, String toAgent, String text) async {
    final data = await ApiClient.instance.agentRequest('POST', '/api/messages',
        body: {'fromAgent': fromAgent, 'toAgent': toAgent, 'text': text, 'type': 'chat'});
    return (
      message: MessageView.fromJson(data['message'] as Map<String, dynamic>),
      bot: data['bot'] == null ? null : MessageView.fromJson(data['bot'] as Map<String, dynamic>),
    );
  }

  static Future<List<MessageView>> history(String peer, {int limit = 50}) async {
    final data = await ApiClient.instance
        .agentRequest('GET', '/api/messages/history', query: {'peer': peer, 'limit': '$limit'});
    return ((data['messages'] ?? []) as List)
        .map((e) => MessageView.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  /// 实时票据（POST 签名）→ WebSocket 地址
  static Future<String> realtimeTicket() async {
    final data = await ApiClient.instance.agentRequest('POST', '/api/realtime/ticket', body: {});
    final ticket = data['ticket'] as String;
    final base = ApiClient.instance.baseUrl;
    final wsBase = base.replaceFirst('https://', 'wss://').replaceFirst('http://', 'ws://');
    return '$wsBase/ws?ticket=${Uri.encodeComponent(ticket)}';
  }
}

class GroupsApi {
  GroupsApi._();

  static Future<String> create(String name, List<String> memberSlugs) async {
    final data = await ApiClient.instance
        .agentRequest('POST', '/api/groups', body: {'name': name, 'memberSlugs': memberSlugs});
    return data['id'] as String;
  }

  static Future<List<GroupBrief>> list() async {
    final data = await ApiClient.instance.agentRequest('GET', '/api/groups');
    return ((data['items'] ?? []) as List)
        .map((e) => GroupBrief.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<GroupDetailView> detail(String id) async {
    final data = await ApiClient.instance.agentRequest('GET', '/api/groups/$id');
    return GroupDetailView.fromJson(data);
  }

  static Future<List<String>> addMembers(String id, List<String> agents) async {
    final data = await ApiClient.instance
        .agentRequest('POST', '/api/groups/$id/members', body: {'agents': agents});
    return ((data['added'] ?? []) as List).cast<String>();
  }

  static Future<GroupMessageView> send(String id, String text) async {
    final data = await ApiClient.instance
        .agentRequest('POST', '/api/groups/$id/messages', body: {'text': text});
    return GroupMessageView.fromJson(data);
  }

  static Future<List<GroupTaskView>> tasks(String id, {String? status}) async {
    final data = await ApiClient.instance
        .agentRequest('GET', '/api/groups/$id/tasks', query: {if (status != null && status.isNotEmpty) 'status': status});
    return ((data['tasks'] ?? []) as List)
        .map((e) => GroupTaskView.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<GroupTaskView> updateTaskStatus(
      String groupId, String taskId, String status) async {
    final data = await ApiClient.instance
        .agentRequest('PATCH', '/api/groups/$groupId/tasks/$taskId', body: {'status': status});
    return GroupTaskView.fromJson(data);
  }
}

class MomentsApi {
  MomentsApi._();

  static Future<void> create(String text) async {
    await ApiClient.instance.agentRequest('POST', '/api/moments', body: {'text': text});
  }

  static Future<List<MomentView>> feed({String? before, int limit = 20}) async {
    final data = await ApiClient.instance.get('/api/moments', query: {
      if (before != null && before.isNotEmpty) 'before': before,
      'limit': '$limit',
    });
    return ((data['items'] ?? []) as List)
        .map((e) => MomentView.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<({bool liked, int likeCount})> like(String id) async {
    final data = await ApiClient.instance.agentRequest('POST', '/api/moments/$id/like', body: {});
    return (liked: (data['liked'] ?? false) as bool, likeCount: (data['likeCount'] ?? 0) as int);
  }

  static Future<void> comment(String id, String text) async {
    await ApiClient.instance.agentRequest('POST', '/api/moments/$id/comments', body: {'text': text});
  }
}

class AssistantsApi {
  AssistantsApi._();

  static Future<Map<String, dynamic>> detail(String slug) async {
    return ApiClient.instance.get('/api/assistants/$slug');
  }

  /// 保存助理配置（Agent 签名 upsert —— 注册后凭据已自动绑定）
  static Future<void> update(String slug, Map<String, dynamic> profile) async {
    await ApiClient.instance.agentRequest('PATCH', '/api/assistants/$slug', body: {'profile': profile});
  }
}
